django/authentication/serializers.py

```diff
@@ -0,0 +1,116 @@
+from rest_framework import serializers
+from .models import User
+from django.contrib import auth
+from rest_framework.exceptions import AuthenticationFailed
+from django.contrib.auth.tokens import PasswordResetTokenGenerator
+from django.utils.encoding import smart_str, force_str, smart_bytes, DjangoUnicodeDecodeError
+from django.utils.http import urlsafe_base64_decode, urlsafe_base64_encode
+from django.contrib.sites.shortcuts import get_current_site
+from django.urls import reverse
+from .utils import Util
+
+#We import force_str in order to retreive the human readable id
+class RegisterSerializer(serializers.ModelSerializer):#Added Registeration Serializer, with conditions
+    password = serializers.CharField(
+        max_length=255, min_length=8, write_only=True)
+
+    class Meta:
+        model = User
+        fields = ['username', 'email', 'password'
+                  ]
+
+    def validate(self, attrs):
+        email = attrs.get('email', '')
+        username = attrs.get('username', '')
+        if not username.isalnum():
+            raise serializers.ValidationError('This username should only contain alphanumeric characters')
+        return attrs
+    
+    def create(self, validated_data):
+        return User.objects.create_user(**validated_data)
+
+class EmailVerificationSerializer(serializers.ModelSerializer): #Email Verification Serializer
+    token = serializers.CharField(max_length=555)
+
+    class Meta:
+        model = User
+        fields = ['token']
+
+class LoginSerializer(serializers.ModelSerializer):# Created Login Serializer Linking refresh tokens from jwt
+    email = serializers.EmailField(
+        max_length=255, min_length=8)
+    password = serializers.CharField(
+        max_length=68, min_length=3, write_only=True)
+    username = serializers.CharField(
+        max_length=255, min_length=3, read_only=True)
+    tokens = serializers.CharField(
+        max_length=68, min_length=3,read_only=True)
```

Comment:
Is there a need for these specific min and max length values? If so, the source should be documented. Otherwise, in my experience all min/max length constraints should be set in the model and not in the serializer. Also, I believe that the email, password, and username fields all will be inherited automatically from the `User` model, so we should not need to add them manually here.